### ISSUE TITLE
Yum module reports OK despite package not installed

### ISSUE THREAD

**Issue by henriklynggaard** [ansible/ansible-modules-core#5411]
##### ISSUE TYPE
- Bug Report
  ##### COMPONENT NAME
  Ansible core module: yum 
##### ANSIBLE VERSION

```
vagrant@localhost aon_ansible]$ ansible-playbook --version
ansible-playbook 2.1.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

Have not made any changes, ansible installed via fedora 24 package manager as "dnf ansible"
##### OS / ENVIRONMENT

Fedora 24 Workstation
[vagrant@localhost aon_ansible]$ uname -a
Linux localhost.localdomain 4.5.5-300.fc24.x86_64 #1 SMP Thu May 19 13:05:32 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
##### SUMMARY

Using yum module to install several packagges with_items, I get 
TASK [Install Docker Engine] ***************************************************
ok: [10.115.141.233] => (item=[u'device-mapper-libs', u'device-mapper-event-libs', u'docker'])
despite docker package is not available
PLAY RECAP *********************************************************************
10.115.141.232             : ok=1    changed=0    unreachable=0    failed=1   
##### STEPS TO REPRODUCE

Example play:
  tasks:
- name: Install Docker Engine
  yum: name={{item}} state=installed
  with_items:
  - device-mapper-libs
  - device-mapper-event-libs
  - docker

Running playbook
[vagrant@localhost aon_ansible]$ ansible-playbook -i inventories/test
playbooks/docker-provision.yml 

PLAY [ams_docker] **************************************************************

TASK [setup] *******************************************************************
ok: [10.115.141.232]
ok: [10.115.141.233]

TASK [Install Docker Engine] ***************************************************
ok: [10.115.141.232] => (item=[u'device-mapper-libs', u'device-mapper-event-libs', u'docker'])
ok: [10.115.141.233] => (item=[u'device-mapper-libs', u'device-mapper-event-libs', u'docker'])

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @/home/vagrant/Source/aon_ansible/playbooks/docker-provision.retry

PLAY RECAP *********************************************************************
10.115.141.232             : ok=1    changed=0    unreachable=0    failed=1  
10.115.141.233             : ok=1    changed=0    unreachable=0    failed=1  

Running with verbose:
[vagrant@localhost aon_ansible]$ ansible-playbook -v -i inventories/test  playbooks/docker-provision.yml 
Using /etc/ansible/ansible.cfg as config file

PLAY [ams_docker] **************************************************************

TASK [setup] *******************************************************************
ok: [10.115.141.233]
ok: [10.115.141.232]

TASK [Install Docker Engine] ***************************************************
ok: [10.115.141.233] => (item=[u'device-mapper-libs', u'device-mapper-event-libs', u'docker']) => {"changed": false, "failed": true, "item": ["device-mapper-libs", "device-mapper-event-libs", "docker"], "msg": "No Package matching 'docker' found available, installed or updated", "rc": 0, "results": ["device-mapper-libs-1.02.107-5.el7_2.5.x86_64 providing device-mapper-libs is already installed", "device-mapper-event-libs-1.02.107-5.el7_2.5.x86_64 providing device-mapper-event-libs is already installed"]}
ok: [10.115.141.232] => (item=[u'device-mapper-libs', u'device-mapper-event-libs', u'docker']) => {"changed": false, "failed": true, "item": ["device-mapper-libs", "device-mapper-event-libs", "docker"], "msg": "No Package matching 'docker' found available, installed or updated", "rc": 0, "results": ["device-mapper-libs-1.02.107-5.el7_2.5.x86_64 providing device-mapper-libs is already installed", "device-mapper-event-libs-1.02.107-5.el7_2.5.x86_64 providing device-mapper-event-libs is already installed"]}

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @/home/vagrant/Source/aon_ansible/playbooks/docker-provision.retry

PLAY RECAP *********************************************************************
10.115.141.232             : ok=1    changed=0    unreachable=0    failed=1  
10.115.141.233             : ok=1    changed=0    unreachable=0    failed=1 
##### EXPECTED RESULTS

Summary would not show "OK" on task level and show error message
##### ACTUAL RESULTS

Task shows OK 

```
TASK [Install Docker Engine] ***************************************************
ok: [10.115.141.233] => (item=[u'device-mapper-libs', u'device-mapper-event-libs', u'docker']) => {"changed": false, "failed": true, "item": ["device-mapper-libs", "device-mapper-event-libs", "docker"], "msg": "No Package matching 'docker' found available, installed or updated", "rc": 0, "results": ["device-mapper-libs-1.02.107-5.el7_2.5.x86_64 providing device-mapper-libs is already installed", "device-mapper-event-libs-1.02.107-5.el7_2.5.x86_64 providing device-mapper-event-libs is already installed"]}
ok: [10.115.141.232] => (item=[u'device-mapper-libs', u'device-mapper-event-libs', u'docker']) => {"changed": false, "failed": true, "item": ["device-mapper-libs", "device-mapper-event-libs", "docker"], "msg": "No Package matching 'docker' found available, installed or updated", "rc": 0, "results": ["device-mapper-libs-1.02.107-5.el7_2.5.x86_64 providing device-mapper-libs is already installed", "device-mapper-event-libs-1.02.107-5.el7_2.5.x86_64 providing device-mapper-event-libs is already installed"]}

```

Note that inner message has {"changed": false, "failed": true, "msg": "No Package matching 'docker' found available, installed or updated")

**Comment by ansibot** [comment]
@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jtognazzi** [comment]
I also noticed this bug see [here](https://groups.google.com/d/msg/ansible-project/q2ylOE2usPw/VrPbyw9xBgAJ)

seems to be a duplicate of #5228

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ssbarnea** [comment]
This bug was fixed on 2.2.1 release.
